sync/pooled_worker_pool.go

```diff
@@ -73,14 +73,43 @@ func (p *pooledWorkerPool) Go(work Work) {
 	workCh <- work
 }
 
-func (p *pooledWorkerPool) spawnWorker(workCh chan Work) {
+func (p *pooledWorkerPool) GoOrGrow(work Work) {
+	// Use time.Now() to avoid excessive synchronization
+	workChIdx := p.nowFn().UnixNano() % p.numShards
+	workCh := p.workChs[workChIdx]
```

Comment:
Maybe move this code into a minor reused fn:
```
func (p *pooledWorkerPool) selectWorkCh() chan Work {
  workChIdx := p.nowFn().UnixNano() % p.numShards
  return p.workChs[workChIdx]
}
```

Reply:
sure